python/lsst/ts/watcher/model.py

```diff
@@ -230,6 +230,37 @@ def get_rules(self, name_regex):
         compiled_re = re.compile(name_regex)
         return [rule for name, rule in self.rules.items() if compiled_re.match(name) is not None]
 
+    def mute_alarm(self, name, duration, severity, user):
+        """Mute one or more alarms for a specified duration.
+
+        Parameters
+        ----------
+        name : `str`
+            Alarm name.
+            If a regular expression then acknowledge all matching alarms.
+        duration : `float`
+            How long to mute the alarm (sec)
+        severity : `lsst.ts.idl.enums.Watcher.AlarmSeverity` or `int`
+            Severity to acknowledge. If the severity goes above
+            this level the alarm will unacknowledge itself.
+        user : `str`
+            Name of user; used to set acknowledged_by.
+        """
+        for rule in self.get_rules(name):
+            rule.alarm.mute(duration=duration, severity=severity, user=user)
+
+    def unmute_alarm(self, name):
+        """Unmute one or more alarms.
+
+        Parameters
+        ----------
+        name : `str`
+            Alarm name. If "all" then unmute all alarms.
```

Comment:
Instead of "all", wouldn't "*." suffice, given that its a regex? Is there special logic for the "all" case? If so, I didn't see it.

Reply:
Outdated info. I switched to using "*" for the reasons you gave. Fixed.